### PR TITLE
Issues - Forms / Auth Redirects - [RET-287] [RET-341] [RET-344] [RET-345] [RET-346] [RET-347]

### PR DESCRIPTION
**Pull request by mhfen**
### What's this PR do?
- Fixes issues with refs in login form.
- Fixes issues with server errors in login form.
- Refactors Auth Route handling to use React Router `<Redirect />`
- Pulls out some code we were not using anymore

### Where should the reviewer start?
`app.jsx` and `routes.jsx` have the changes for the Router
`login.jsx` has the changes for the forms

### Does this add new dependencies? Is there an installation procedure? (bundle, bower, npm, etc)
No.

### What are the relevant tickets?
https://mondorobot.atlassian.net/browse/RET-287
https://mondorobot.atlassian.net/browse/RET-341
https://mondorobot.atlassian.net/browse/RET-344
https://mondorobot.atlassian.net/browse/RET-345
https://mondorobot.atlassian.net/browse/RET-346
https://mondorobot.atlassian.net/browse/RET-347

### Who should review this PR?
@cprater @mhfen

### CODE REVIEW

File: src/client/components/dropdown.jsx [hunk 1]
```diff
@@ -45,12 +45,21 @@ export default class Dropdown extends React.Component {
...
+      'isOpen': !this.state.isOpen,
+    }, this.toggleOffClickListener);
+
+    this.setTriggerText(option);
+  }
+
+  setTriggerText(option) {
     const triggerTxt = option.description ? option.description : this.state.triggerTxt;
 
+    if (!this.props.dynamicTrigger) return;
```
Comment (mhfen) [inline src/client/components/dropdown.jsx]: @cprater I changed `Dropdown` to allow some to have static triggers. Updated your Session dropdowns to have the `dynamicTrigger` prop.

File: src/client/pages/sessions/sessions.jsx [hunk 2]
```diff
@@ -195,21 +195,23 @@ export default class Sessions extends Component {
             <div className="block">
               <div className="block-item">
                 <Dropdown
-                  animate="bottomLeft"
+                  animate="bottomRight"
```
Comment (cprater) [inline src/client/pages/sessions/sessions.jsx]: I think these do want to be `bottomLeft`

File: src/client/components/dropdown.jsx (hunk 1 above)
Comment (cprater) [inline src/client/components/dropdown.jsx]: sweet 👍

File: src/client/pages/sessions/sessions.jsx (hunk 2 above)
Comment (cprater) [inline src/client/pages/sessions/sessions.jsx]: NVM

**Pull request merged**